packages/ui/cypress/tests/setIdentity.cy.ts

```diff
@@ -0,0 +1,105 @@
+    sendTxModal.sendTxTitle().should('be.visible')
+    sendTxModal.setIdentitySection().should('be.visible')
+
+    // every field is empty to start with
+    sendTxModal.setIdentityFieldInput('display').should('have.value', '')
+    sendTxModal.setIdentityFieldInput('legal').should('have.value', '')
+
+    // typing into legal with empty display should show an error
+    sendTxModal.setIdentityField('legal').type('le')
+    sendTxModal.sendTxError().should('contain', 'Display name is required')
+    sendTxModal.setIdentityFieldElement('display', 'label').should('havInpute.class', 'Mui-error')
+    sendTxModal.setIdentityFieldElement('display', 'div').should('havInpute.class', 'Mui-error')
+    sendTxModal.buttonSend().should('be.disabled')
+
+    // typing into display should remove the error
+    sendTxModal.setIdentityField('display').type('diis')
+    sendTxModal.sendTxError().should('not.exist')
+    sendTxModal.setIdentityFieldElement('display', 'label').should('not.have.class', 'Mui-error')
+    sendTxModal.buttonSend().should('be.enabled')
+
+    // typing a too long field should show another error
+    sendTxModal.setIdentityField('legal').type('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')
+    sendTxModal.sendTxError().should('contain', 'A field exceeds the 32 character limit')
+    sendTxModal.setIdentityFieldElement('legal', 'label').should('have.class', 'Mui-error')
+    sendTxModal.buttonSend().should('be.disabled')
+
+    // removing should remove the error
+    sendTxModal.setIdentityField('legal').type('{selectall}{del}leeg')
+    sendTxModal.sendTxError().should('not.exist')
+    sendTxModal.setIdentityFieldElement('legal', 'label').should('not.have.class', 'Mui-error')
+    sendTxModal.buttonSend().should('be.enabled')
+
+    // verify the tx it sends
+    sendTxModal.buttonSend().click()
+    waitForTxRequest()
+    cy.getTxRequests().then((req) => {
+      const txRequests = Object.values(req)
+      cy.wrap(txRequests.length).should('eq', 1)
+      cy.wrap(txRequests[0].payload.address).should('eq', multisigSignatoryWithoutIdentity.address)
+      sendTxModal.buttonSend().should('be.disabled')
+      // expected https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/extrinsics/decode/0x1f0102000412ad770206045069201514711dc2456908b0af226442d475d12a5334e9c4513e001901000564696973056c6565670000000000000000
+      cy.wrap(txRequests[0].payload.method).should(
+        'eq',
+        '0x1f0102000412ad770206045069201514711dc2456908b0af226442d475d12a5334e9c4513e001901000564696973056c6565670000000000000000'
+      )
+    })
+  })
+
+  it.only('Can edit an identity from the new tx button', () => {
+    const multisigSignatoryWithoutIdentity = setIdentitySignatories[0]
+    initAndConnect(multisigSignatoryWithoutIdentity)
+    // select the right multisig (Alice has a lot)
+    const first5DigitsAddress = setIdentityMultisigs['multisig-with-identity'].address.slice(0, 4)
```

Comment:
Just a thought - we might have to use this a lot, maybe it's declaring on the the test fixtures?

Reply:
as a helper function eventually, but I think in the fixture it'd be bloating it maybe?

Reply:
Thinking about it, I think it's too simple and the code to get it will be just as long as repeating `slice(0, 4)` Because you'd end up with `getFirst5Char(setIdentityMultisigs['multisig-with-identity'].address)` which to me is the same, with an import spared :shrug:

Reply:
Haha fair point! 👍